Numsense/Chinese.fs

```diff
@@ -0,0 +1,194 @@
+
+open Ploeh.Numsense.InternalDsl
+
+type CharacterSet =
+    {
+        negative: string;
+        zero: string;
+        one: string;
+        two: string;
+        alternativeTwo: string;
+        three: string;
+        four: string;
+        five: string;
+        six: string;
+        seven: string;
+        eight: string;
+        nine: string;
+        ten: string;
+        hundred: string;
+        thousand: string;
+        tenThousands: string;
+        hundredMillions: string
+    }
+
+let internal TraditionalCharacterSet =
+    {
+        negative = "負";
+        zero = "零";
+        one = "一";
+        two = "二";
+        alternativeTwo = "兩";
+        three = "三";
+        four = "四";
+        five = "五";
+        six = "六";
+        seven = "七";
+        eight = "八";
+        nine = "九";
+        ten = "十";
+        hundred = "百";
+        thousand = "千";
+        tenThousands = "萬";
+        hundredMillions = "億"
+    }
+
+let internal TraditionalFinancialCharacterSet =
+    {
+        negative = "負";
+        zero = "零";
+        one = "壹";
+        two = "貳";
+        alternativeTwo = "兩";
+        three = "叄";
+        four = "肆";
+        five = "伍";
+        six = "陸";
+        seven = "柒";
+        eight = "捌";
+        nine = "玖";
+        ten = "拾";
+        hundred = "佰";
+        thousand = "仟";
+        tenThousands = "萬";
+        hundredMillions = "億"
+    }
+
+let internal SimplifiedCharacterSet =
+    {
+        negative = "负";
+        zero = "零";
+        one = "一";
+        two = "二";
+        alternativeTwo = "两";
+        three = "三";
+        four = "四";
+        five = "五";
+        six = "六";
+        seven = "七";
+        eight = "八";
+        nine = "九";
+        ten = "十";
+        hundred = "百";
+        thousand = "千";
+        tenThousands = "万";
+        hundredMillions = "亿"
+    }
+
+let internal SimplifiedFinancialCharacterSet =
+    {
+        negative = "负";
+        zero = "零";
+        one = "壹";
+        two = "贰";
+        alternativeTwo = "两";
+        three = "叁";
+        four = "肆";
+        five = "伍";
+        six = "陆";
+        seven = "柒";
+        eight = "捌";
+        nine = "玖";
+        ten = "拾";
+        hundred = "佰";
+        thousand = "仟";
+        tenThousands = "万";
+        hundredMillions = "亿"
+    }
+
+let rec internal toChineseImp characterSet x =
+
+    let getCharacter useAlternativeTwo x =
+        match x with
+        | 0 -> characterSet.zero
+        | 1 -> characterSet.one
+        | 2 -> if useAlternativeTwo then characterSet.alternativeTwo else characterSet.two
+        | 3 -> characterSet.three
+        | 4 -> characterSet.four
+        | 5 -> characterSet.five
+        | 6 -> characterSet.six
+        | 7 -> characterSet.seven
+        | 8 -> characterSet.eight
+        | 9 -> characterSet.nine
+        | _ -> ""
+
+    let formatTenThousand suffix previousValue x =
+        let formatInternal suffix mayOutputZero isQuantity x =
```

Comment:
Have you considered refactoring this nested `if` expression to pattern matching on the tuple `x, mayOutputZero`?